language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/SymbolUtil.java

```diff
@@ -218,6 +219,25 @@ public static boolean isRecordVariable(Symbol variableSymbol) {
         }
     }
 
+    /**
+     * Get the type of the symbol. For functions, this returns the return type of the function symbol.
+     *
+     * @param symbol symbol of which the type is required
+     * @return Optional type of the provided symbol
+     */
+    public static Optional<TypeSymbol> getReturnTypeDescriptor(Symbol symbol) {
+        if (symbol == null) {
+            return Optional.empty();
+        }
+
+        if (symbol.kind() == SymbolKind.FUNCTION || symbol.kind() == SymbolKind.METHOD) {
+            FunctionTypeSymbol functionTypeSymbol = ((FunctionSymbol) symbol).typeDescriptor();
+            return functionTypeSymbol.returnTypeDescriptor();
+        }
+
+        return SymbolUtil.getTypeDescriptor(symbol);
```

Comment:
Agreed to the fact that this has an issue with the meaning. I created this because the same logic is used in both `Variable Declarations` and `Assignments` related completion providers. What do we do in that case?

Reply:
If we look at the `getTypeDescriptor` there we return the `FunctionSymbol` for Functions/ Methods. I think therefore we can logically structure this to first call the `getTypeDescriptor` method and then to the line 233-235 evaluation. Also, renaming this would improve the usage

Reply:
Yes. I will update to do so.

Reply:
Updated. Can you check?